### PR TITLE
feature(DH): Add more contact information

### PR DESCRIPTION
**Pull request by Angi-Kinas**
This PR adds more contact information in the metadata contact section if it exists.

It will look like this:
image

image

image

### CODE REVIEW

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.spec.ts
Comment (cmoinier) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.spec.ts:114]: Forgot to delete this console.log :slightly_smiling_face:

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts
Comment (cmoinier) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts:38]: The only thing that worries me is if one address doesn't have any commas (as we spoke about). Is this a case that can happen? @fgravin

**Review by cmoinier — CHANGES_REQUESTED** [review]
Thanks for the work :slightly_smiling_face: 
I tested and it works well with all of my datasets.

Only UI thing to fix IMO is the spacing between the address and the email in case there is no website, i.e. without website (too narrow) :

image

and with website : 

image

I also would like to wait for a second opinion about the risk of having addresses without commas (if there is no risk then I will approve the PR) :slightly_smiling_face:

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (fgravin) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:49]: Please use built-in classes like `w-5` instead, when possible.

**Review by fgravin — COMMENTED** [review]
A similar work has been started in https://github.com/geonetwork/geonetwork-ui/pull/575
Dunno if it helps.

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:36]:
```suggestion
  <div *ngIf="contacts[0]?.phone" class="mb-5">
```

Otherwise JS errors pop in case of a missing contact

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:47]:
```suggestion
  <div *ngIf="contacts[0]?.email" class="mb-5">
```

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:64]:
```suggestion
  <div *ngIf="contacts[0]?.firstName || contacts[0]?.lastName" class="mb-5">
```

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:72]: This might render weirdly if only one of the two is defined

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:77]:
```suggestion
  <div *ngIf="contacts[0]?.address" class="mb-2">
```

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:6]: this needs an `overflow-hidden` class otherwise there is a "cut edges" effect on logos, see:
image

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html:90]: why would we need to split the logic here? having `ngFor` loop on all parts would be enough I guess

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts
Comment (jahow) [reply libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts:38]: yes definitely, there's no rule for addresses as far as I know; but then it'll simply end up being a long line, so we just need to make sure that the text overflow works well :) (looks to be the case in my tests)

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts
Comment (jahow) [inline libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts:58]: I'm not sure about this logic; it makes a lot of assumptions about the structure of an address, and honestly I don't think we can assume that much as address fields can often be filled completely freely by metadata authors; e.g. assuming that "CS" will be present before a postcode (which is specific to France I guess), assuming that the city name comes before the postal code...

Relying on the "," separator is enough IMO, and putting each part in a separate line.

**Review by jahow — COMMENTED** [review]
Added some more comments, thanks for the contribution!

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts
Comment (jahow) [reply libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts:58]: On the other hand it looks like the GN editor reliably produces this kind of output... but that might change anytime in the future

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts
Comment (Angi-Kinas) [reply libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts]: Okay in this case I will keep it simple as you were proposing @jahow and only split by ',' and if that's not there, it will be a long line of text which overflows, as you said. I implemented it to fit the design, but I guess there are too many different variations of the user input.

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (Angi-Kinas) [reply libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html]: I think now it should be handled correctly.

File: libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html
Comment (Angi-Kinas) [reply libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.html]: Good catch! Thanks :)

**Review by jahow — APPROVED** [review]
Thanks for the improvements! As discussed, the layout can probably be slightly improved using `grid` and `gap` instead of `mb` classes. I'm approving this, please merge when you feel this is ready!

**Pull request merged**
